src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -213,7 +236,16 @@ void PiecewisePolynomial<MaxDeriv>::pup(PUP::er& p) {
     } else if (version >= 2) {
       // However, for v2+, we store expiration time, size, then data for
       // thread-safety reasons
-      p | expiration_time_;
+
+      // For v3+ the expiration time is atomic. While for 2 it isn't
+      if (version >= 3) {
+        double loaded_expiration_time = 0.0;
+        p | loaded_expiration_time;
+        expiration_time_.store(loaded_expiration_time,
+                               std::memory_order_relaxed);
```

Comment:
Pup `operator|` is defined for `std::atomic` in `Utilities/Serialization/PupStlCpp11.hpp`.

It's also compatible with the serialization of the non-atomic contained type, so you don't need to bump the version.